### PR TITLE
Hpc best practices

### PR DESCRIPTION
**Pull request by b10n1k**
- Use of Mojo::Base https://docs.mojolicious.org/Mojo/Base
      strict and warnings are inherited by this Class
- Use kernel maillist for maintainer                                                                                                                                                                                                 
                                                                                                                                                                                                                                             
With this PR all the HPC modules introduce the `signatures` flag which is                                                                                                                                                                    
supported by `Mojo::Base`. The signature declares lexical variables that are                                                                                                                                                                 
in scope for the block                                                                                                                                                                                                                       
and every subroutine invocation, first is checked for its signature.


- Related ticket: https://progress.opensuse.org/issues/xyz
- Verification run: 
https://openqa.suse.de/tests/8266698
https://openqa.suse.de/tests/8266713#dependencies

### CODE REVIEW

File: lib/hpc/cluster.pm [hunk 1]
```diff
@@ -1,24 +1,21 @@
 # SUSE's openQA tests
 #
-# Copyright 2020 SUSE LLC
+# Copyright 2020-2022 SUSE LLC
```
Comment (okurz) [inline lib/hpc/cluster.pm:3]: don't update copyright years with ranges, see https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/CONTRIBUTING.md?plain=1#L49

File: tests/hpc/ganglia_client.pm [hunk 2]
```diff
@@ -1,23 +1,18 @@
 # SUSE's openQA tests
 #
-# Copyright 2018 SUSE LLC
+# Copyright 2018-2022 SUSE LLC
 # SPDX-License-Identifier: FSFAP
 
 # Summary: Ganglia Test - client
 #   Acts as client node, which publishes data to the server via gmetric command
-# Maintainer: soulofdestiny <mgriessmeier@suse.com>
-# Tags: https://fate.suse.com/323979
```
Comment (okurz) [inline tests/hpc/ganglia_client.pm:9]: I suggest to keep that. Even if fate is not actively used anymore it still acts as a reference explaining why we have test modules at all

**Review by okurz — CHANGES_REQUESTED** [review]

File: lib/hpc/cluster.pm (hunk 1 above)
Comment (b10n1k) [reply lib/hpc/cluster.pm]: *this change is ready for review*

File: tests/hpc/ganglia_client.pm (hunk 2 above)
Comment (b10n1k) [reply tests/hpc/ganglia_client.pm:9]: *this change is ready for review*

**Review by okurz — APPROVED** [review]
Please await review from former and updated test module maintainers

**Comment by schlad** [comment]
nice cleanup! thx @b10n1k

**Comment by dzedro** [comment]
https://openqa.suse.de/tests/8284597

**Pull request merged**